src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class IsoscelesTrapezoid extends Figure {
+    public static final int MAXIMUM_SIZE = 50;
+    private int bottom;
+    private int top;
+    private int height;
+    private Random random = new Random();
+
+    @Override
+    public void randomizeProperty() {
+        bottom = random.nextInt(MAXIMUM_SIZE);
+        top = random.nextInt(MAXIMUM_SIZE);
+        height = random.nextInt(MAXIMUM_SIZE);
+    }
```

Comment:
The randomizeProperty method is generating random values inside the figure class which contradicts the checklist point that suggests generating random values in the FigureSupplier. Consider refactoring this to pass random values to the constructor.